subsys/zigbee/cli/zigbee_cli_cmd_bdb.c

```diff
@@ -0,0 +1,864 @@
+	legacy_mode = ZB_TRUE;
+
+	if (ZB_SCHEDULE_APP_CALLBACK(zb_bdb_set_legacy_device_support,
+				     legacy_mode)) {
+		zb_cli_print_error(shell, "Can not execute command", ZB_FALSE);
+		return -ENOEXEC;
+	}
+
+	zb_cli_print_done(shell, ZB_FALSE);
+	return 0;
+}
+
+/**@brief Disable the legacy device support.
+ *
+ * @code
+ * bdb legacy disable
+ * @endcode
+ *
+ * Disallow legacy pre-r21 devices on the Zigbee network.
+ *
+ * @pre Use only after @ref start "bdb start".
+ *
+ * Example:
+ * @code
+ * > bdb legacy disable
+ * Done
+ * @endcode
+ */
+static int cmd_zb_legacy_disable(const struct shell *shell, size_t argc,
+				 char **argv)
+{
+	if (!zigbee_is_stack_started()) {
+		zb_cli_print_error(shell, "Stack not started", ZB_FALSE);
+		return -ENOEXEC;
+	}
+
+	if (zb_get_network_role() != ZB_NWK_DEVICE_TYPE_COORDINATOR) {
+		zb_cli_print_error(shell, "Device must be a coordinator",
+				   ZB_FALSE);
+		return -ENOEXEC;
+	}
+
+	zb_bdb_set_legacy_device_support(0);
+	legacy_mode = ZB_FALSE;
+
+	if (ZB_SCHEDULE_APP_CALLBACK(zb_bdb_set_legacy_device_support,
+				     legacy_mode)) {
+		zb_cli_print_error(shell, "Can not execute command", ZB_FALSE);
+		return -ENOEXEC;
+	}
+
+	zb_cli_print_done(shell, ZB_FALSE);
+	return 0;
+}
+
+/**@brief Set network key.
+ *
+ * @code
+ * bdb nwkkey <h:key>
+ * @endcode
+ *
+ * Set a pre-defined network key instead of a random one.
+ *
+ * @pre Setting only before @ref start "bdb start".
+ *
+ * Example:
+ * @code
+ * > bdb nwkkey 00112233445566778899aabbccddeeff
+ * Done
+ * @endcode
+ */
+static int cmd_zb_nwkkey(const struct shell *shell, size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+
+	if (zigbee_is_stack_started()) {
+		zb_cli_print_error(shell, "Stack already started", ZB_FALSE);
+		return -ENOEXEC;
+	}
+
+	zb_uint8_t key[ZB_CCM_KEY_SIZE];
+
+	if (parse_hex_str(argv[1], strlen(argv[1]), key, sizeof(key), false)) {
+		zb_secur_setup_nwk_key(key, 0);
+	} else {
+		zb_cli_print_error(shell, "Failed to parse key", ZB_FALSE);
+		return -EINVAL;
+	}
+
+	zb_cli_print_done(shell, ZB_FALSE);
+	return 0;
+}
+
+/**@brief Perform a factory reset via local action
+ *
+ * See Base Device Behavior specification chapter 9.5 for details.
+ *
+ * @code
+ * > bdb factory_reset
+ * Done
+ * @endcode
+ */
+static int cmd_zb_factory_reset(const struct shell *shell, size_t argc,
+				char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	ZB_SCHEDULE_APP_CALLBACK(zb_bdb_reset_via_local_action, 0);
+	zb_cli_print_done(shell, ZB_FALSE);
+	return 0;
+}
+
+/**@brief Set amount of the child devices which can be connected to the device.
+ *
+ * @code
+ * bdb child_max <d:children_nbr>
+ * @endcode
+ *
+ *  @pre Setting only before @ref start "bdb start".
+ *
+ * If the argument is provided and `children_nbr` is in [0:32] range,
+ * set to that number. Otherwise, return error.
+ *
+ *
+ * Example:
+ * @code
+ * > bdb child_max 16
+ * Setting max children to: 16
+ * Done
+ * @endcode
+ */
+#ifndef ZB_ED_ROLE
```

Comment:
In the future please use the macro:
`SHELL_COND_CMD_ARG_REGISTER` to register such command. It won't create a command in `ZB_ED_ROLE` is not defined.